model/facade/action/GetPersonalSummaryByLoginDateAction.php

```diff
@@ -76,6 +77,42 @@ public function __construct(UserVO $userVO, DateTime $date) {
 
     }
 
+    /**
+     * @return mixed
+     * @throws null
+     */
+    private function getWorkableHoursInThisYear() {
+        $firstDayOfYear = date_create($this->date->format('Y-1-01'));
+        $lastDayOfYear = date_create($this->date->format('Y-12-31'));
+
+        //Now need to find out the workable hours in this year
+        $extraHoursAction = new ExtraHoursReportAction($firstDayOfYear, $lastDayOfYear, $this->userVO);
+        $results = $extraHoursAction->execute();
+        return $results[1][$this->userVO->getLogin()]['workable_hours'];
+
+    }
+
+    /**
+     * @return mixed
+     * @throws null
+     */
+    private function getWorkedHoursInThisYear() {
+        $firstDayOfYear = date_create($this->date->format('Y-1-01'));
+
+        // We need to get the number of hours worked till date
+        $extraHoursAction = new ExtraHoursReportAction($firstDayOfYear, $this->date, $this->userVO);
+        $results = $extraHoursAction->execute();
+        return $results[1][$this->userVO->getLogin()]['total_hours'];
```

Comment:
It's better if you modify ExtraHoursReportAction to take an optional $user parameter. That's a quite heavy report and you will be wasting resources if you calculate the report for every user and then discard them.